client/src/lib/paymentCalc.ts

```diff
@@ -1,56 +1,117 @@
 import { ClientConstants } from "../constants";
-import { AbiType, AxiomV2ClientOverrides, AxiomV2QueryOptions } from "../types";
+import { AbiType, AxiomV2ClientOverrides, AxiomV2ClientOptions } from "../types";
 import { PublicClient } from "viem";
-import { getAxiomV2QueryAddress } from "./address";
-import { getAxiomV2Abi } from "./abi";
+import { getAxiomV2QueryAddress, getOpStackGasPriceOracleAddress } from "./address";
+import { getAxiomV2Abi, getOpStackGasPriceOracleAbi } from "./abi";
+import { isArbitrumChain, isMainnetChain, isOpStackChain, isScrollChain } from "./chain";
+import { publicActionsL2 } from 'viem/op-stack';
+import { readContractValue } from "./viem";
 
 export async function calculatePayment(
-  axiomV2QueryAddr: string,
+  chainId: string,
   publicClient: PublicClient,
-  options?: AxiomV2QueryOptions
+  options: AxiomV2ClientOptions,
 ): Promise<bigint> {
+  const axiomV2QueryAddr = options.overrides?.queryAddress ?? getAxiomV2QueryAddress(chainId);
+
+  // Get callback gas limit
+  const callbackGasLimit = BigInt(options.callbackGasLimit ?? ClientConstants.DEFAULT_CALLBACK_GAS_LIMIT);
+
+  // Get maxFeePerGas
+  const maxFeePerGas = BigInt(options.maxFeePerGas ?? ClientConstants.DEFAULT_MAX_FEE_PER_GAS_WEI);
```

Comment:
Just realized -- we'll want to make the ClientConstants depend on `chainId`, e.g. this `DEFAULT_MAX_FEE_PER_GAS_WEI` should be quite different on Base / mainnet